utility/src/test/java/org/oppia/android/util/profile/NameValidatorTest.kt

```diff
@@ -0,0 +1,36 @@
+package org.oppia.android.util.profile
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import javax.inject.Inject
+
+class NameValidatorTest {
+
+  @Inject
+  lateinit var nameValidator: NameValidator
+
+  private val allowedNames = listOf<String>("जिष्णु", "Ben-Henning", "Rajat.T", "جيشنو")
+
+  private val disallowedNames =
+    listOf<String>("जिष्णु7", "Ben_Henning", "Rajat..T", "جيشنو^&&", " ", ".", "Ben Henning")
+
+  @Before
+  fun setup() {
+    nameValidator = NameValidator()
+  }
+
+  @Test
+  fun testNameValidator_addDisallowedName_returnFalse() {
+    disallowedNames.forEach {
+      assertThat(nameValidator.isNameValid(it)).isEqualTo(false)
+    }
+  }
+
+  @Test
+  fun testNameValidator_addAllowedName_returnTrue() {
```

Comment:
Rather than organizing these tests as pass/fail, you should structure them based on the specific behavior that's being tested, the condition that triggers that behavior, and subsequent outcome. Each behavior should be in its own test. For example:

``testIsNameValid_nameWithSpaces_returnsFalse``

or

```testNameIsValid_nameHasOneDash_returnsTrue``

etc.

Reply:
Added. PTAL and let me know if more tests are needed